backend/endpoints/rooms.py

```diff
@@ -170,3 +170,95 @@ async def join_room(
         status_code=status.HTTP_424_FAILED_DEPENDENCY,
         detail="failed to add new members to room",
     )
+
+
+@router.put(
+    "/org/{org_id}/rooms/{room_id}",
+    response_model=ResponseModel,
+    status_code=status.HTTP_200_OK,
+    responses={
+        401: {"detail": "member not in room"},
+        403: {"detail": "cannot close a channel room"},
+        404: {"detail": "room not found"},
+        424: {"detail": "unable to close conversation"},
+    },
+)
+async def close_conversation(
+    org_id: str,
+    room_id: str,
+    member_id: str,
+    background_tasks: BackgroundTasks,
+):
+    """
+    Closes a DM or Group_DM room on the sidebar
+    By toggling the closed boolean field of the room document from False to True
+    The function when called on a closed room, changes the closed field back to False
+    Args:
+        org_id (str): A unique identifier of an organisation
+        room_id: A unique identifier of the room to be updated
+        member_id: A unique identifier of the member initiating the request
+        background_tasks: A parameter that allows tasks to be performed outside of the main function
+    Returns:
+        HTTP_200_OK: {
+                        "status": "success",
+                        "message": "conversation closed || conversation opened",
+                        "data": {
+                            "closed": true || false,
+                            "role": "admin",
+                            "starred": false
+                        }
+                    }
+    Raises:
+        HTTP_401_UNAUTHORIZED: member not in room
+        HTTP_403_FORBIDDEN: cannot close a channel room
+        HTTP_404_NOT_FOUND: room not found
+        HTTP_424_FAILED_DEPENDENCY: failed to add new members to room
+    """
+    DB = DataStorage(org_id)
+    room = await get_room(org_id=org_id, room_id=room_id)
+
+    if not room:
+        raise HTTPException(
+            status_code=status.HTTP_404_NOT_FOUND, detail="room not found"
+        )
+
+    if room["room_type"].upper() == RoomType.CHANNEL:
+        raise HTTPException(
+            status_code=status.HTTP_403_FORBIDDEN, detail="cannot close a channel room"
+        )
+
+    if member_id not in room["room_members"].keys():
+        raise HTTPException(
+            status_code=status.HTTP_401_UNAUTHORIZED, detail="member not in room"
+        )
+
+    member_room_data = room["room_members"].get(member_id)
+    member_room_data["closed"] = member_room_data["closed"] is False
+    data = {"room_members": room["room_members"]}
+
+    update_response = await DB.update(
+        ROOM_COLLECTION, document_id=room_id, data=data
+    )  # updates the room data in the db collection
+
+    background_tasks.add_task(
+        centrifugo_client.publish,
+        room=room_id,
```

Comment:
Is everyone in the room to be notified of a closed room?
Thought it's peculiar to the user who closed the room?

Reply:
no everyone is not supposed to be notified. does the code suggest that?
the publish is just supposed to do a sidebar update so that the sidebar details for that particular member will be updated so that the room can be excluded.
Please tell me if it seems different @AI-fae

Reply:
the notification is just for that particular member, does the  code suggest otherwise? please let me know. 
the sidebar update publish is for updating the sidebar details of that particular member so that the closed room is removed from the joined rooms list.

Reply:
publishing to the room id will publish to every member of the room if I remember correctly.

If you want to update just the user's sidebar, you might want to publish to the user id instead. Ref the create_room endpoint to see how this was done.

Reply:
In the sidebar.py file, there's a helper function for updating the sidebar on centrifugo. You should take a look at it too...``

Reply:
I'm not sure an event can be published to a user_id. 
But I will take a look at the suggested files and see what can be done. Thanks